core/src/main/java/com/linecorp/armeria/server/docs/DocServicePlugin.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright 2017 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceConfig;
+
+/**
+ * Generates the {@link ServiceSpecification}s of the supported {@link Service}s.
+ */
+public interface DocServicePlugin {
+
+    // Methods related with generating a ServiceSpecification.
+
+    /**
+     * Returns the {@link Set} of the {@link Service} types supported by this plugin.
+     */
+    Set<Class<? extends Service<?, ?>>> supportedServiceTypes();
+
+    /**
+     * Generates a new {@link ServiceSpecification} that provides the information about the supported
+     * {@link Service}s.
+     *
+     * @param serviceConfigs the {@link ServiceConfig}s of the {@link Service}s that are instances of the
+     *                       {@link #supportedServiceTypes()}
+     */
+    ServiceSpecification generateSpecification(Set<ServiceConfig> serviceConfigs);
+
+    // Methods related with extracting documentation strings.
+    // TODO(trustin): Define the docstring format.
+    // TODO(trustin): How do we specify the docstring of an exception thrown by a method?
```

Comment:
I think both of these are usually in the method docstring itself, which I guess is fine as is.